### ISSUE TITLE
Is there an edge-case lurking in our complaint-handling logic?

### ISSUE THREAD

**Issue by philbooth** [mozilla/fxa-auth-server#2450]
[In `lib/email/bounces.js`](https://github.com/mozilla/fxa-auth-server/blob/44e9dacc9d7abfb9c12e9fe55c286b3fe053be8f/lib/email/bounces.js#L51-L55), we treat `complainedRecipients` as equivalent to `bouncedRecipients` in its meaning:

```js
if (message.bounce) {
  recipients = message.bounce.bouncedRecipients
} else if (message.complaint) {
  recipients = message.complaint.complainedRecipients
}
```

However, [in the SES docs](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/notification-contents.html#complained-recipients) an important distinction is made:

> **Important**
>
> Most ISPs remove the email address of the recipient who submitted the complaint from their complaint notification. For this reason, this list contains information about recipients who might have sent the complaint, based on the recipients of the original message and the ISP from which we received the complaint. Amazon SES performs a lookup against the original message to determine this recipient list.

In practice, there's not much harm in this for us right now because we only send email to one user at a time. The worst thing that could happen is we mark more than one of a user's email addresses as complained when actually only one of them did. But given that we're only sending email to their verified addresses (right?), the repercussions of this seem small (if someone is complaining from one email address, they probably want to complain for all of them).

It may prove to be more problematic for fxa-email-service though, if/when that repo becomes a proper standalone service open to reliers. In the presence of messages that are sent to multiple recipients at the same email provider, we will not be able to apportion the complaint to a specific recipient.

So, in summary, this is all very low-risk at the moment and probably not actionable anyway. But I wanted to give it some visibility ahead of triage today, we can close it if you guys agree with most of what I've written above.

**Comment by rfk** [comment]
>  if you guys agree with most of what I've written above.

I agree with most of what you've written above.

**Comment by philbooth** [comment]
Closing this issue, it's served its purpose. mozilla/fxa-email-service#26 can stay open for the somewhat thornier problem of how we might need to handle it in the future.
